bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromBulbHandler.java

```diff
@@ -0,0 +1,314 @@
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.core.cache.ExpiringCache;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link MyStromBulbHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frederic Chastagnol - Initial contribution
+ */
+@NonNullByDefault
+public class MyStromBulbHandler extends MyStromAbstractHandler {
+
+    private static class MyStromBulbResponse {
+        public boolean on;
+        public String color = "";
+        public String mode = "";
+        public long ramp;
+
+        @Override
+        public String toString() {
+            return "MyStromBulbResponse{" + "on=" + on + ", color='" + color + '\'' + ", mode='" + mode + '\''
+                    + ", ramp=" + ramp + '}';
+        }
+    }
+
+    private static class MyStromDeviceSpecificInfo extends MyStromBulbResponse {
+        public double power;
+    }
+
+    private static final String COMMUNICATION_ERROR = "Error while communicating to the myStrom bulb: ";
+
+    private final Logger logger = LoggerFactory.getLogger(MyStromBulbHandler.class);
+
+    private final ExpiringCache<Map<String, MyStromDeviceSpecificInfo>> cache = new ExpiringCache<>(
+            Duration.ofSeconds(3), this::getReport);
+
+    private PercentType lastBrightness = PercentType.HUNDRED;
+    private QuantityType<Dimensionless> lastColorTemperature = QuantityType.ONE;
+
+    public MyStromBulbHandler(Thing thing, HttpClient httpClient) {
+        super(thing, httpClient);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                pollDevice();
+            } else {
+                @Nullable
+                String sResp = null;
+                switch (channelUID.getId()) {
+                    case CHANNEL_SWITCH:
+                        if (command instanceof OnOffType) {
+                            sResp = sendHttpPost(command == OnOffType.ON ? "on" : "off", null, null, null);
+                        }
+                        break;
+
```

Comment:
check SAT report and remove unnecessary empty lines.

Reply:
The generated html files in target directory? They show neither errors nor warnings.

Reply:
Running install issues the following warning:
```
Jan. 23, 2021 5:16:26 NACHM. org.openhab.tools.analysis.checkstyle.KarafFeatureCheck processFiltered
WARNUNG: Could not find file feature file features/karaf/openhab-addons/src/main/feature/feature.xml
```
I have no idea what it is and how and whether I have to correct anything.

Reply:
You can ignore that. It's a bug in the static analysis tool.